packages/contractkit/src/providers/celo-private-keys-subprovider.ts

```diff
@@ -178,15 +198,47 @@ export class CeloPrivateKeysWalletProvider extends PrivateKeyWalletSubprovider {
     return this.gasFeeRecipient
   }
 
-  private async getGasPrice(): Promise<string> {
-    debug('getGasPrice fetching gas price...')
+  private async getGasPrice(gasCurrency: string | undefined): Promise<string | undefined> {
+    // Gold Token
+    if (!gasCurrency) {
+      return this.getGasPriceInCeloGold()
+    }
+    if (!this.gasPriceMinimumContractPromise) {
+      throw new Error(
+        `celo-private-keys-subprovider@getGasPrice: gas price for ` +
+          `currency ${gasCurrency} cannot be computed since GasPriceMinimum contract is not missing`
+      )
+    }
+    const gasPriceMinimumContract = await this.gasPriceMinimumContractPromise
```

Comment:
still not sure about this...

First, setting a gas price is usually something the user does by itself. Since it wants to decide wether he wants a fast, normal or slow transaction.  There's a website and api for that: https://ethgasstation.info

So, it wouldn't be that bad if we don't support this here yet.

Second, even if kit doesn't yet support this. There are a few things that need to be handled:

 * contract address might change due governance
 * contract impl might change due governance

kit will eventually react to this; and invalidate actual contracts.

Third, abstraction wise, a provider should not be aware (dependant) on web3, or a contract from it. If he needs to talks to the network, it can use `emitPayload()`. In that sense, it could:

a) know the address
b) encodeDataABI for a call to getGasPriceMinimun
c) do a low level call using `emitPayload` calling `eth_call` rpc method

Reply:
As per the discussion, removed gas limit estimation. Hard-coded it to 90,000 which is the default value in Geth.